pretty_midi/pretty_midi.py

```diff
@@ -117,6 +146,38 @@ def _load_tempo_changes(self, midi_data):
                     if tick_scale != last_tick_scale:
                         self.__tick_scales.append((event.tick, tick_scale))
 
+    def _load_metadata(self, midi_data):
+        """Populates the lists of tempo key and time signature changes
+
+        Populates self.__time_signatures with TimeSignature objects.
+            It can be accessed by the get_time_signatures method.
+        Populates self.__key_changes with KeySignature objects.
+            It can be accessed by the get_key_signatures method.
+
+        Parameters
+        ----------
+            - midi_data : midi.FileReader
+                MIDI object from which data will be read
+        """
+
+        # list to store key signature changes
+        self.__key_changes = []
```

Comment:
Well, all of them would basically just need to check that the thing being added is an instance of what the list is supposed to hold.  Usually in Python you let the user shoot themselves in the foot as far as properties go, and avoid getter/setters whenever possible; see e.g. http://stackoverflow.com/questions/1554546/when-and-how-to-use-the-builtin-function-property-in-python

If you really believe we need getter/setters with sanity checks, we can do them but only as properties, and we'd need them for all of the list attributes.